planner/core/stats.go

```diff
@@ -549,6 +547,12 @@ func (ds *DataSource) buildIndexMergeOrPath(partialPaths []*util.AccessPath, cur
 	indexMergePath := &util.AccessPath{PartialIndexPaths: partialPaths}
 	indexMergePath.TableFilters = append(indexMergePath.TableFilters, ds.pushedDownConds[:current]...)
 	indexMergePath.TableFilters = append(indexMergePath.TableFilters, ds.pushedDownConds[current+1:]...)
+	for _, path := range partialPaths {
+		if len(path.TableFilters) > 0 {
+			indexMergePath.TableFilters = append(indexMergePath.TableFilters, ds.pushedDownConds[current])
```

Comment:
Should we append the entire DNF in the `pushedDownConds` or just the `path.TableFilters`?

Reply:
`path.TableFilters` is enough, since `ds.pushedDownConds[current] = path.TableFilters + path.AccessConds` and `path.AccessConds` will take effect when scan. @time-and-fate